src/minio/managers/user_manager.py

```diff
@@ -265,6 +265,44 @@ async def get_user(self, username: str) -> UserModel:
                 accessible_paths=sorted(list(all_accessible_paths)),
             )
 
+    async def get_or_rotate_user_credentials(self, username: str) -> Tuple[str, str]:
+        """
+        Generate fresh credentials for a user by rotating their password/secret key.
+
+        This method implements a unified credential system where:
+        - Access key is always the username (consistent identifier)
+        - Secret key is a freshly generated secure password
+        - The MinIO user password is updated to match the new secret key
+
+        Args:
+            username: The username to generate fresh credentials for
+        """
+        async with self.operation_context("get_or_rotate_user_credentials"):
+
+            user_exists = await self.resource_exists(username)
+            if not user_exists:
+                raise UserOperationError(f"User {username} not found")
```

Comment:
You could move this to right before the _excecutor line to reduce the chance of race conditions

Reply:
👍 sure.